cmake/modules/ErrorFlags.cmake

```diff
@@ -65,8 +68,12 @@ elseif(MSVC)
 			"/WX" # Treat warnings as errors
 		)
 	endif()
+
+	set(SILENCE_DEPRECATION_DEFS "_SILENCE_CXX20_OLD_SHARED_PTR_ATOMIC_SUPPORT_DEPRECATION_WARNING")
+	set(THIRD_PARTY_SILENCE_DEPRECATION_DEFS "")
 endif()
 
 # Add the flags to the whole directory tree. We use the third-party flags for
 # targets whose SYSTEM property is true, and the normal flags otherwise.
 add_compile_options("$<IF:$<BOOL:$<TARGET_PROPERTY:SYSTEM>>,${THIRD_PARTY_COMPILE_ERROR_FLAGS},${COMPILE_ERROR_FLAGS}>")
+add_compile_definitions("$<IF:$<BOOL:$<TARGET_PROPERTY:SYSTEM>>,${THIRD_PARTY_SILENCE_DEPRECATION_DEFS},${SILENCE_DEPRECATION_DEFS}>")
```

Comment:
I don't think we'll ever need `THIRD_PARTY_SILENCE_DEPRECATION_DEFS` - there shouldn't be any deprecation warnings to silence, as we turn off warnings entirely for third-party code. Similarly, adding the first-party definitions to third-party code should have no effect if the warnings are off anyway. Thus, I think the changes to this file can be simplified to one line by adding the definition unconditionally in the MSVC block.

I think it would also be worth adding a comment that these changes (the MSVC definition and whatever GCC flag is required to address Johannes' comment) are temporary and, once we can fully update to C++20, should be removed in favour of fixing the code.

Reply:
Done